OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -109,14 +123,18 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 			if (order.OrderString == "BeginMinefield")
 				minefieldStart = cell;
 			else if (order.OrderString == "PlaceMine")
-				self.QueueActivity(order.Queued, new LayMines(self));
+			{
+				if (IsCellAcceptable(self, cell))
+					self.QueueActivity(order.Queued, new LayMines(self));
+			}
 			else if (order.OrderString == "PlaceMinefield")
 			{
 				var movement = self.Trait<IPositionable>();
+				var mobile = movement as Mobile;
 
 				var minefield = GetMinefieldCells(minefieldStart, cell, Info.MinefieldDepth)
-					.Where(c => movement.CanEnterCell(c, null, BlockedByActor.Immovable)
-						|| (!self.Owner.Shroud.IsVisible(c) && self.World.Map.Contains(c)))
+					.Where(c => self.World.Map.Contains(c) && IsCellAcceptable(self, c) && self.Owner.Shroud.IsExplored(c)
```

Comment:
Shouldn't `self.World.Map.Contains(c)` be part of the internals of `IsCellAcceptable`? I'd expect `GetTerrainInfo` to throw or return bogus results for cells outside of the map.